### PR TITLE
Add kotlin sample code in "Integration with an Android Fragment" documentation

### PR DESCRIPTION
**Pull request by mdvacca**
This PR updates the "[https://reactnative.dev/docs/integration-with-android-fragment](Integration with an Android Fragment)" documentation to display Kotlin and java code samples

This is the first step on the effort to display Kotlin code in RN OSS documentation

integration-with-android-fragment-kotlin

### CODE REVIEW

File: website/core/TabsConstants.js [hunk 1]
```diff
@@ -19,6 +19,13 @@ const packageManagers = [
 ];
 const defaultPackageManager = 'npm';
 
+const androidLanguages = [
+  {label: 'Java', value: 'java'},
+  {label: 'Kotlin', value: 'kotlin'},
+];
+
+const defaultAndroidLanguage = 'kotlin';
```
Comment (ShikaSD) [inline website/core/TabsConstants.js:27]: Are we sure the default should be Kotlin?

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ id: integration-with-android-fragment
 title: Integration with an Android Fragment
 ---
 
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
```
Comment (ShikaSD) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:6]: nit: can imports be multiline?

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 3]
```diff
@@ -17,12 +19,58 @@ You will need to implement the ReactApplication interface in your main Applicati
...
+
+```kotlin
+class MyReactApplication: Application(), ReactApplication {
+
+    override fun onCreate(): Unit {
+        super.onCreate()
+        SoLoader.init(this, false)
+    }
+
+    private val reactNativeHost = object: ReactNativeHost(this) {
```
Comment (ShikaSD) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:52]:
```suggestion
    private val reactNativeHost = object : ReactNativeHost(this) {
```

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 4]
```diff
@@ -101,12 +173,51 @@ Now in your Activity class e.g. `MainActivity.java` you need to add an OnClickLi
...
 
+<Tabs groupId="android-language" defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
+<TabItem value="kotlin">
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle): Unit {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.main_activity)
+
+    button = findViewById(R.id.button) as Button
```
Comment (ShikaSD) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:203]: I am pretty sure Android has a generic option now?
```suggestion
    button = findViewById<Button>(R.id.button)
```

You probably don't even need the `<>` specification, but not sure.

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md (hunk 2 above)
Comment (mdvacca) [reply website/versioned_docs/version-0.67/integration-with-android-fragment.md:6]: I followed the same format used by others doc pages. can we include this as part of another PR where we make them all consistent?

File: website/core/TabsConstants.js (hunk 1 above)
Comment (mdvacca) [reply website/core/TabsConstants.js]: fixed it

**Review by Simek — CHANGES_REQUESTED** [review]
Hello @mdvacca, thank you for the updates! 👍 That's great idea to adopt more Kotlin code alternative examples on the website.

I have only two small request:
* for some reason, it looks like for tabs text the `value` is used, not the `label`, which leads to the lowercased tab captions
  Screenshot 2022-03-03 172619
* the edits have been performed only on the versioned docs, this means that the changes will apply only for the 0.67 docs and won't be populated for the future versions, the source for versioned docs generation is `/docs` folder, and it content is also available under the `Next` tag on the website. In this situation I would recommend to apply the same changes to the `/docs` files and leave the 0.67 docs file as is, because 0.67 it is currently the default version, so it will make the changes more accessible (there will be no need to change to `Next` to see the Kotlin examples). 🙂

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 5]
```diff
@@ -17,12 +19,58 @@ You will need to implement the ReactApplication interface in your main Applicati
...
+        override fun getPackages(): List<ReactPackage> {
+            val packages = PackageList(this).getPackages()
+            // Packages that cannot be autolinked yet can be added manually here
+            return packages
+        }
+    }
+
+    override fun getReactNativeHost(): ReactNativeHost {
+        return reactNativeHost
+    }
```
Comment (cortinico) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:67]:
```suggestion
    override fun getReactNativeHost(): ReactNativeHost = reactNativeHost
```

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 6]
```diff
@@ -17,12 +19,58 @@ You will need to implement the ReactApplication interface in your main Applicati
...
+
 Override the required methods `getUseDeveloperSupport`, `getPackages` and `getReactNativeHost`:
 
+<Tabs groupId="android-language" defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
+<TabItem value="kotlin">
+
+```kotlin
+class MyReactApplication: Application(), ReactApplication {
+
+    override fun onCreate(): Unit {
```
Comment (cortinico) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:47]:
```suggestion
    override fun onCreate() {
```

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 7]
```diff
@@ -51,8 +99,29 @@ public class MyReactApplication extends Application implements ReactApplication
...
+```kotlin
+import android.app.Application
+
+import com.facebook.react.PackageList
+import com.facebook.react.ReactApplication
+import com.facebook.react.ReactNativeHost
+import com.facebook.react.ReactPackage
+import com.facebook.soloader.SoLoader
+
+import java.util.List
```
Comment (cortinico) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:119]: This should be removed. When on Kotlin you should use Kotlin std-lib's type wrappers.

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 8]
```diff
@@ -101,12 +173,51 @@ Now in your Activity class e.g. `MainActivity.java` you need to add an OnClickLi
...
+</TabItem>
+</Tabs>
+
 Update your Activity's onCreate method as follows:
 
+<Tabs groupId="android-language" defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
+<TabItem value="kotlin">
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle): Unit {
```
Comment (cortinico) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:199]:
```suggestion
override fun onCreate(savedInstanceState: Bundle) {
```

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 9]
```diff
@@ -131,12 +242,30 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
+<Tabs groupId="android-language" defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
+<TabItem value="kotlin">
+
+```kotlin
+private fun getLaunchOptions(String message): Bundle {
+    val initialProperties = Bundle()
+    initialProperties.putString("message", message)
+    return initialProperties
+}
```
Comment (cortinico) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:262]: This doesn't compile as it is (`String message` -> `message: String`).

Anyway I would reword to:

```suggestion
private fun getLaunchOptions(message: String) = Bundle().apply {
    putString("message", message)
}
```

**Review by cortinico — COMMENTED** [review]
Great stuff 🚀 It's awesome to see more Koltin code in the React Native docs!

Left several comments on how to improve the code

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md (hunk 2 above)
Comment (Simek) [reply website/versioned_docs/version-0.67/integration-with-android-fragment.md:6]: @ShikaSD @mdvacca IIRC in the past Docusuaurus throw an compile error when import were listed on the separate lines, but maybe it's not the case anymore. I agree, that it would be better to correct this formatting if possible, I will try to change this during next Docusuaurs bump, which will be soon. 🙂

Comment (Simek) [reply website/versioned_docs/version-0.67/integration-with-android-fragment.md:6]: It looks like that this is still an issue, but the Docusuaurs team has found a workaround, which is wrapping the imports in `mdx-code-bock`:
* https://github.com/facebook/docusaurus/blame/main/website/versioned_docs/version-2.0.0-beta.17/guides/markdown-features/markdown-features-react.mdx#L10-L15

**Comment by Simek** [comment]
> for some reason, it looks like for tabs text the value is used, not the label, which leads to the lowercased tab captions

It looks like, the missing `export`s for newly added `androidLanaguages` are the cause here. We probably should rewrite the file, so we use inline exports (`export const = ...`) instead just to avoid issues like this. 😅 

Also I'm not sure why Docusaurus did not warn about this problem, but maybe it is due to the way in which imports are used in MD files.

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 10]
```diff
@@ -17,12 +19,54 @@ You will need to implement the ReactApplication interface in your main Applicati
...
+        super.onCreate()
+        SoLoader.init(this, false)
+    }
+
+    private val reactNativeHost = object : ReactNativeHost(this) {
+
+        override fun getUseDeveloperSupport() = BuildConfig.DEBUG
+
+        override fun getPackages(): List<ReactPackage> {
+            val packages = PackageList(this).getPackages()
```
Comment (ShikaSD) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:57]: does `.getPackages()` return `List` or `MutableList`? Assuming the intention here was to allow manual additions here, it won't work in Kotlin because `List` doesn't have `.add` methods

**Review by ShikaSD — APPROVED** [review]
One small nit, looks good to me otherwise

**Review by mdvacca — COMMENTED** [review]
updated getPackages() method

**Comment by cortinico** [comment]
Code looks good to me, but it needs to be reformatted with:
https://facebookincubator.github.io/ktfmt/

I tried to leave code suggestions to this PR, but I think I broke Github ¯\_(ツ)_/¯ Whenever I try to post a comment with a code suggestion I receive a 500 from GH, not sure why.

We can merge it as it is and reformat the code afterwards, or maybe @mdvacca you can take a stab and reformat those snippets using the Web UI of Ktfmt

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 11]
```diff
@@ -17,12 +19,54 @@ You will need to implement the ReactApplication interface in your main Applicati
...
+
+        override fun getPackages(): List<ReactPackage> {
+            val packages = PackageList(this).getPackages().toMutableList()
+            // Packages that cannot be autolinked yet can be added manually here
+            return packages
+        }
+    }
+
+    override fun getReactNativeHost(): ReactNativeHost = reactNativeHost
+}
```
Comment (cortinico) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:64]:
```suggestion
class MyReactApplication : Application(), ReactApplication {
    override fun onCreate() {
        super.onCreate()
        SoLoader.init(this, false)
    }
    private val reactNativeHost =
        object : ReactNativeHost(this) {
            override fun getUseDeveloperSupport() = BuildConfig.DEBUG
            override fun getPackages(): List<ReactPackage> {
                val packages = PackageList(this).getPackages().toMutableList()
                // Packages that cannot be autolinked yet can be added manually here
                return packages
            }
        }
    override fun getReactNativeHost(): ReactNativeHost = reactNativeHost
}
```

File: website/versioned_docs/version-0.67/integration-with-android-fragment.md [hunk 12]
```diff
@@ -101,12 +167,51 @@ Now in your Activity class e.g. `MainActivity.java` you need to add an OnClickLi
...
+                .setComponentName("HelloWorld")
+                .setLaunchOptions(getLaunchOptions("test message"))
+                .build()
+
+        getSupportFragmentManager()
+                .beginTransaction()
+                .add(R.id.reactNativeFragment, reactNativeFragment)
+                .commit()
+    }
+}
```
Comment (cortinico) [inline website/versioned_docs/version-0.67/integration-with-android-fragment.md:209]:
```suggestion
override fun onCreate(savedInstanceState: Bundle) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.main_activity)
    
    button = findViewById<Button>(R.id.button)
    button.setOnClickListener {
        val reactNativeFragment =
            ReactFragment.Builder()
                .setComponentName("HelloWorld")
                .setLaunchOptions(getLaunchOptions("test message"))
                .build()
        getSupportFragmentManager()
            .beginTransaction()
            .add(R.id.reactNativeFragment, reactNativeFragment)
            .commit()
    }
}
```

**Comment by mdvacca** [comment]
Thanks @cortinico, I just accepted/committed all your code suggestions

**Review by cortinico — APPROVED** [review]
